### PR TITLE
feat(layout): Refractoring

### LINKED ISSUE #6922: The table of contents does not work for translated pages for community package managers
### URL:

https://nodejs.org/fr/download/package-manager/all

### Browser Name:

Firefox

### Browser Version:

127.0.2

### Operating System:

Ubuntu 22.04

### How to reproduce the issue:

1. Click on the link `Debian et Distributions Linux basées sur Ubuntu` in the table of content
2. See that it does not scroll to the content

The table of content uses `#debian-and-ubuntu-based-linux-distributions` as anchor (using the English anchor) while the header uses `distributions-basées-sur-debian-et-ubuntu` as id (translating the anchor)

Some other anchors in that table of content are also impacted

### PR DESCRIPTION
**Pull request by AugustinMauroy**
## Description

- Remove handwritten ToC, and use `metabar`
- Clean and merge some layout to simplify our logic.

## Related Issues

It's one solution for #6922 

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npm run format` to ensure the code follows the style guide.
- [x] I have run `npm run test` to check if all tests are passing.
- [x] I have run `npx turbo build` to check if the website builds without errors.

### CODE REVIEW

**Comment by ovflowd** [comment]
Is this still a draft?

**Comment by AugustinMauroy** [comment]
> Is this still a draft?

Yes and no. Basically, the component and file names are clearly not the right ones. But I have no idea.
But from a code point of view nothing will change in my opinion

**Comment by bmuenzenmeyer** [comment]
Is the only thing blocking this the naming?

**Comment by AugustinMauroy** [comment]
Yeah 😅 I'm not fan about how layout is named

**Comment by mikeesto** [comment]
ArticlePageLayout?

**Comment by bmuenzenmeyer** [comment]
I'm working on this addition to the COLLABORATOR_GUIDE. I was hoping it would help us identify the `SimplePage` uniqueness, and perhaps simplify our layouts - we have a LOT

#### Layouts

There are several Layouts available within the Website. Here's a comparison of their features:

| Layout       | Includes Layout... | `WithNavBar` | `WithSidebar` | `WithFooter` | `WithMetaBar` | `WithBreadcrumbs` | Comment / Custom Element(s)                          |
| ------------ | ------------------ | ------------ | ------------- | ------------ | ------------- | ----------------- | ---------------------------------------------------- |
| `Default`    | `Article`          | ✅           | ✅            | ✅           | ❌            | ❌                |                                                      |
| `About`      | `Article`          | ✅           | ✅            | ✅           | ✅            | ✅                |                                                      |
| `SimplePage` | `Article`          | ✅           | ✅            | ❌           | ❌            | ❌                |                                                      |
| `Learn`      | `Article`          | ✅           | ❌            | ✅           | ✅            | ✅                | `WithSidebarCrossLinks`                              |
| `Post`       | `Content`          | ✅           | ❌            | ✅           | ✅            | ❌                | `WithBlogCrossLinks`                                 |
| `Blog`       |                    | ✅           | ❌            | ✅           | ❌            | ❌                | `WithBlogCategories`                                 |
| `Download`   |                    | ✅           | ❌            | ✅           | ❌            | ❌                | `WithDownloadCategories`                             |
| `Search`     |                    | ✅           | ❌            | ✅           | ❌            | ❌                |                                                      |
| `Centered`   |                    | ✅           | ❌            | ❌           | ❌            | ❌                |                                                      |
| `Base`       |                    | ❌           | ❌            | ❌           | ❌            | ❌                | `NotificationProvider` and `NavigationStateProvider` |
| `Home`       | `Centered`         | ❌           | ❌            | ❌           | ❌            | ❌                | `GlowingBackdrop`                                    |
| `Article`    |                    | ❌           | ❌            | ❌           | ❌            | ❌                | CSS Only                                             |
| `Content`    |                    | ❌           | ❌            | ❌           | ❌            | ❌                | CSS Only                                             |

**Comment by AugustinMauroy** [comment]
Centered and home can be merge; and content with post ?

**Comment by ovflowd** [comment]
> Centered and home can be merge; and content with post ?

If it works, it works. Do it.

File: apps/site/layouts/Home.tsx [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
-  <CenteredLayout>
-    <GlowingBackdrop />
+  <>
+    <WithNavBar />
+    <div className={styles.centeredLayout}>
+      <GlowingBackdrop />
 
-    <main className={styles.homeLayout}>{children}</main>
-  </CenteredLayout>
+      <main className={styles.homeLayout}>{children}</main>
```
Comment (AugustinMauroy) [inline apps/site/layouts/Home.tsx]: Augustin self review: this break the 404 and error layout (flex-row ≠ flex-col)

**Comment by ovflowd** [comment]
cc @AugustinMauroy can you rebase this branch, it has the old bug of D.map is not a function on Vercel

File: apps/site/pages/en/download/package-manager/all.md [hunk 2]
```diff
@@ -1,42 +1,12 @@
 ---
-layout: download
```
Comment (canerakdas) [inline apps/site/pages/en/download/package-manager/all.md:2]: I am not sure if we have solved the table of contents issue on the `/download/package-manager/all` page right way. Although there is a download page, accessing other download pages is not as easy as before. In addition, on mobile, the table of contents is too end of the page for this layout. Wouldn't it be better to solve this problem with the current layout (download)? 🤔

**Comment by canerakdas** [comment]
This PR solves more than one problem (#7062, #6922) and simplifies the layouts, do you have time to resolve the conflict @AugustinMauroy 🙇 

I have availability this week and I can help too if you want 👀

**Comment by AugustinMauroy** [comment]
@canerakdas it's not overcomplicated just need to open spotify 😁

**Review by canerakdas — APPROVED** [review]
LGTM 🎵 I checked the layouts in general for desktop and mobile resolution. Thank you @AugustinMauroy

**Pull request merged**